app/src/main/java/io/pslab/activity/CompassActivity.java

```diff
@@ -295,6 +297,9 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 bottomSheetBehavior.setState(bottomSheetBehavior.getState() == BottomSheetBehavior.STATE_HIDDEN ?
                         BottomSheetBehavior.STATE_EXPANDED : BottomSheetBehavior.STATE_HIDDEN);
                 break;
+            case android.R.id.home:
+                this.finish();
+                return true;
```

Comment:
Replace these `return true`s with `break`s

Reply:
@CloudyPadmal I've made the changes as per requested.